calnex/firmware/ossfirmware.go

```diff
@@ -31,12 +32,34 @@ type OSSFW struct {
 
 // NewOSSFW returns initialized version of OSSFW
 func NewOSSFW(source string) (*OSSFW, error) {
+	var err error
+	var vs string
+	var v *version.Version
+
 	fw := &OSSFW{
 		filepath: source,
 	}
 	basename := filepath.Base(fw.filepath)
-	vs := strings.ReplaceAll(strings.TrimSuffix(basename, filepath.Ext(basename)), "sentinel_fw_v", "")
-	v, err := version.NewVersion(strings.ToLower(vs))
+	// Extract version from filename
+	// sentinel_fw_v2.13.1.0.5583D-20210924.tar
+	// calnex_combined_fw_R21.0.0.9705-20241111.tar
+	if strings.HasPrefix(basename, "sentinel_fw_") {
+		vs = strings.ReplaceAll(strings.TrimSuffix(basename, filepath.Ext(basename)), "sentinel_fw_", "")
+	} else if strings.HasPrefix(basename, "sentry_fw_") {
+		vs = strings.ReplaceAll(strings.TrimSuffix(basename, filepath.Ext(basename)), "sentry_fw_", "")
+	} else if strings.HasPrefix(basename, "calnex_combined_fw_") {
+		vs = strings.ReplaceAll(strings.TrimSuffix(basename, filepath.Ext(basename)), "calnex_combined_fw_", "")
```

Comment:
you could put these prefixes into a regex with or like https://go.dev/play/p/oiqttenXhFO

Reply:
yes much neater. Implemented